### PR TITLE
Disable Mobile Simulator

### LINKED ISSUE #784: CircuitVerse Doesn't Work on Mobile Devices
**Describe the bug**
On mobile devices, you can't use the simulator!

**To Reproduce**
1. Go to CircuitVerse on a mobile device
2. Go to the simulator
3. Try to use it
4. You'll have a hard time using it

OR

1. Go to CircuitVerse on your browser
2. Inspect Element
3. Change Size to be similar to that of a mobile device
4. Go to the simulator
5. Try to use it
6. You'll have a hard time using the simulator

**Expected behavior**
I expect that using the simulator on any device should be easy and effortless. However, so much as trying to drag an input onto the circuit is frustrating and incredibly difficult.

**Screenshots**
I can't really provide screenshots.

**Desktop (please complete the following information):**
 - OS: Android (But this issue should be across all mobile devices)
 - Browser: Chrome
 - Version: 79.0

**Additional context**
No additional context.

**Are you working on this issue? (Yes/No)**
No.

### PR DESCRIPTION
**Pull request by MrBartusek**
Fixes #784

#### Describe the changes you have made in this pr -
When user open simulator on a mobile device she or he is redirected to new page `/simulator/mobile_disabled`
This function to work require history management in a browser which is [widely supported](https://caniuse.com/#feat=history) but checking for that is also in place
### Screenshots of the changes (If any) -
image

### CODE REVIEW

File: app/controllers/simulator_controller.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ def self.policy_class
     ProjectPolicy
   end
 
+  def mobile_disabled
+    render layout: 'application'
```
Comment (JoshHeng) [inline app/controllers/simulator_controller.rb]: Why not just use the double-quoted strings as codeclimate recommends?

File: app/views/simulator/mobile_disabled.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    <div class="col-lg-6 col-md-6 middle-block">
+      <div class="form-container">
+        <h1 class="pb-3">Simulator Disabled</h1>
+        <p>
+        Unfortunately, CirciutVerse simulator is disabled on mobile devices.
+        You can still use it on your computer.
+        We decided to disable it cause it was hard to use on small screens.
+        Sorry for the inconvenience
+        </p>
```
Comment (JoshHeng) [inline app/views/simulator/mobile_disabled.html.erb]: Check grammar (e.g. spelling of `Circuitverse`).

Personally I would recommend the following text:
`Sorry, but the CircuitVerse simulator is currently disabled on this device. Please use a computer with a larger screen to continue.`

File: public/js/disableMobileSimulator.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        window.location.href = '/simulator/mobile_disabled';
+    }
+}
+function loadSimulatorDisabledPage() {
+    if(window.history) {
+        $('#go-back-button').click(function() {
+            window.history.go(-2);
+        });
+        $('#continue-anyway').click(function() {
+            Cookies.set('disableMobileCheck', true);
```
Comment (JoshHeng) [inline public/js/disableMobileSimulator.js]: I don't think we need a `continue anyway` function or cookie - just disable it altogether

File: app/views/simulator/mobile_disabled.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <button id='go-back-button' type="button" class="btn btn-primary btn-lg btn-block">Go back</button>
+        <div class='text-center'><a id='continue-anyway' lass='font-weight-light' style='color:inherit;'>Continue anyway</a><div>
+      </div>
+    </div>
+  </div>
+</div>
+<script src="https://cdn.jsdelivr.net/npm/js-cookie@2/src/js.cookie.min.js"></script>
```
Comment (JoshHeng) [inline app/views/simulator/mobile_disabled.html.erb]: I don't think we need a `continue anyway` function or cookie - just disable it altogether

File: app/controllers/simulator_controller.rb (hunk 1 above)
Comment (MrBartusek) [reply app/controllers/simulator_controller.rb]: Moved to [Slack](https://circuitverse-team.slack.com/archives/CQFAC5L7P/p1578072012001100)

File: app/views/simulator/mobile_disabled.html.erb (hunk 4 above)
Comment (MrBartusek) [reply app/views/simulator/mobile_disabled.html.erb]: In case of bad device detection or emergency fixes we should allow users to pass

**Comment by MrBartusek** [comment]
Extended in #810

**Pull request closed without merging**